src/AgentCommerce/Util/PayPalCartTransformer.php

```diff
@@ -0,0 +1,286 @@
+
+            $availableShippingMethods->add($shippingOption);
+        }
+
+        return $availableShippingMethods;
+    }
+
+    /**
+     * @return array{validationIssues: ValidationIssueCollection, status: string}
+     */
+    public function convertToValidationIssues(Cart $cart, Context $context): array
+    {
+        $status = PayPalCart::VALIDATION_STATUS__VALID;
+        $errors = new ValidationIssueCollection();
+
+        foreach ($cart->getErrors() as $error) {
+            $validationIssue = new ValidationIssue();
+            // TODO: Add some properties
+
+            $errors->add($validationIssue);
+        }
+
+        $restockProducts = [];
+        $productIds = array_filter($cart->getLineItems()->getReferenceIds());
+        if (!empty($productIds)) {
+            $criteria = new Criteria($productIds);
+            $criteria->addFilter(
+                new RangeFilter('stock', [RangeFilter::LTE => 0]),
+                new NotFilter('AND', [new EqualsFilter('restockTime', null)])
+            );
+            $restockProducts = $this->productRepository->search($criteria, $context)->getElements();
+        }
+
+        foreach ($cart->getLineItems() as $lineItem) {
+            $stock = $lineItem->getPayloadValue('stock'); // @phpstan-ignore method.deprecated
+            if ($stock < $lineItem->getQuantity()) {
+                $restockProduct = $restockProducts[(string) $lineItem->getReferencedId()] ?? null;
+                $issue = $this->validationIssues->outOfStock($lineItem, $restockProduct);
+
+                $errors->add($issue);
+            }
+        }
+
+        // Age verification would be "REQUIRES_ADDITIONAL_INFORMATION"
+        if ($errors->count()) {
+            $status = PayPalCart::VALIDATION_STATUS__INVALID;
+        }
+
+        return ['validationIssues' => $errors, 'status' => $status];
+    }
+
+    public function convertCustomer(?CustomerEntity $customerEntity): ?Customer
+    {
+        if (!$customerEntity) {
+            return null;
+        }
+
+        $customer = new Customer();
+        $name = new CustomerName();
+        $name->setGivenName($customerEntity->getFirstName());
+        $name->setSurname($customerEntity->getLastName());
+
+        $customer->setName($name);
+        $customer->setEmailAddress($customerEntity->getEmail());
+
+        $phoneNumber = $customerEntity->getDefaultShippingAddress()?->getPhoneNumber();
+        if ($phoneNumber) {
+            $phone = new Phone();
+            $phone->setPhoneNumber($phoneNumber);
+            $customer->setPhone($phone);
+        }
+
+        return $customer;
+    }
+
+    /**
+     * @template TAddress as Address
+     *
+     * @param class-string<TAddress> $className
+     *
+     * @return TAddress
+     */
+    public function convertAddress(?CustomerAddressEntity $addressEntity, string $className, Context $context): ?Address
+    {
+        if (!$addressEntity || !is_subclass_of($className, Address::class)) {
```

Comment:
phpstan won't let you anyway:
```suggestion
        if (!$addressEntity) {
```